BEIMA.Backend/MongoService/MongoConnector.cs

```diff
@@ -0,0 +1,192 @@
+                return (ObjectId) doc["_id"];
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
+                return null;
+            }
+        }
+
+        /*
+         * Gets a device from the "devices" collection, given an objectID.
+         * Parameter: objectId, corresponds to the "_id" field for a given document inside of MongoDB
+         * Returns: BsonDocument that was requested
+         */
+        public BsonDocument GetDevice(ObjectId objectId)
+        {
+            if (!IsConnected())
+            {
+                throw new Exception("MongoConnector is not currently connected");
+            }
+
+            var filter = Builders<BsonDocument>.Filter.Eq("_id", objectId);
+
+            try
+            {
+                var db = client.GetDatabase(dbName);
+                var devices = db.GetCollection<BsonDocument>(deviceCollection);
+                return devices.Find(filter).FirstOrDefault();
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
+                return null;
+            }
+        }
+
+        /*
+         * Gets a device from the "devices" collection, given an objectID.
+         * Parameter: objectId, corresponds to the "_id" field for a given document inside of MongoDB
+         * Returns: BsonDocument that was requested
+         */
+        public List<BsonDocument> GetAllDevices()
+        {
+            if (!IsConnected())
+            {
+                throw new Exception("MongoConnector is not currently connected");
+            }
+
+            var filter = Builders<BsonDocument>.Filter.Empty;
+
+            try
+            {
+                var db = client.GetDatabase(dbName);
+                var devices = db.GetCollection<BsonDocument>(deviceCollection);
+                var docs = devices.Find(filter).ToList();
+                return docs;
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
+                return null;
+            }
+        }
+
+        /*
+         * Deletes from the "devices" collection, given the objectID.
+         * Parameter: objectId, corresponds to the "_id" field for a given document inside of MongoDB
+         * Returns: true if successful, false if not successful
+         */
+        public bool DeleteDevice(ObjectId objectId)
+        {
+            if (!IsConnected())
+            {
+                throw new Exception("MongoConnector is not currently connected");
+            }
+
+            var filter = Builders<BsonDocument>.Filter.Eq("_id", objectId);
+
+            try
+            {
+                var db = client.GetDatabase(dbName);
+                var devices = db.GetCollection<BsonDocument>(deviceCollection);
+                var result = devices.DeleteOne(filter);
+                return result.DeletedCount > 0;
```

Comment:
I think it would be better to return the list of deleted devices instead?

Reply:
To delete a device, you would need to already have a copy of the object itself to call this method. Returning the deleted device is redundant since we already have a reference to the device. Having it return a boolean makes sanity checking much easier since we can do something like "if(result)".

Reply:
Yeah, I think I was thinking about this being related to using remove() on a list, but given that we actually need to have the device info to use this, returning true/false seems reasonable.

Reply:
I will keep it the same, it will return a boolean. From what I found, delete operations typically don't return anything. This way, at least we can see if it failed or not by seeing if it returned true/false.